### ISSUE TITLE
Scan all forc.toml's in this repo to check if we have any implicit std dependency on git version of sway-lib-std

### ISSUE THREAD

**Issue by kayagokalp** [FuelLabs/sway#5463]
As a special case (as this repo is the home of sway-lib-std), any forc project hosted in this repo should not depend on the sway lib std implicitly. If they do in release PRs such as #5452 we will see a failure in CI. Because forc tries to fetch std versioned with its own version. For example if we are bumping to 0.49.0, forc will search for std by fetching this repo and going to the 0.49.0 tag. The problem is at the time of that PR is open, 0.49.0 won't be available as it will only be available once it is merged and a new tag is created.

So as a solution projects that do not need std should have `implicit-std = false` in their forc.toml and projects that need them should have it explicitly added as a path dependency:

```TOML
...
[dependencies]
std = { path = "../../../sway-lib-std" }
```

IIRC @ironcev did an amazing job to ensure this in test suite, we can create a script out of it to run it in CI to ensure other test projects are not depending on it either.

**Comment by ironcev** [comment]
Yes there is a single module in the test suite harness checking exactly that (and few other points). I assume it shouldn't be to difficult to extract the logic and use it elsewhere.
